Src/Support/Google.Apis.Auth/OAuth2/ServiceAccountCredential.cs

```diff
@@ -344,7 +362,17 @@ private Task<string> GetOrCreateJwtAccessTokenAsync(string authUri)
                 }
                 // Create a new JWT.
                 var expiryUtc = nowUtc + JwtLifetime;
-                var jwtTask = Task.Run(() => CreateJwtAccessToken(authUri, nowUtc, expiryUtc));
+                Task<string> jwtTask;
+
+                if (HasExplicitScopes)
+                {
+                    jwtTask = Task.Run(() => CreateJwtAccessToken(Scopes, nowUtc, expiryUtc));
+                }
+                else
+                {
+                    jwtTask = Task.Run(() => CreateJwtAccessToken(authUri, nowUtc, expiryUtc));
+                }
+                
```

Comment:
I'd consider pushing the conditional down into the lambda expression, which should make it all rather simpler:

```suggestion
                Task<string> jwtTask = Task.Run(() =>
                    HasExplicitScopes
                    ? CreateJwtAccessToken(Scopes, nowUtc, expiryUtc)
                    : CreateJwtAccessToken(authUri, nowUtc, expiryUtc));
```

Reply:
yep, that looks better, thanks!